hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -784,29 +788,103 @@ private AbfsOutputStreamContext populateAbfsOutputStreamContext(
             .build();
   }
 
-  public void createDirectory(final Path path, final FsPermission permission,
+  public void createDirectory(final Path path, final FileSystem.Statistics statistics, final FsPermission permission,
       final FsPermission umask, TracingContext tracingContext)
-      throws AzureBlobFileSystemException {
+          throws IOException {
     try (AbfsPerfInfo perfInfo = startTracking("createDirectory", "createPath")) {
-      boolean isNamespaceEnabled = getIsNamespaceEnabled(tracingContext);
```

Comment:
instead of ifElse:
lets have:
```
if(prefixMode==BLOB){
   //heuristics;
  return;
}
//old heuristics of the DFS
````
will reduce diff.

Reply:
also, this pr needs create pr. so, in case this pr gets merged, it will hit createPath for directory. Can we have config so that even if this pr is merged, it still hits createDirectory API of dfs endpoint.

Reply:
We would be needing PR's in order to be merged to work as expected. Config would be needed to be rolled back once createPR is merged.